pkgs/development/tools/build-managers/bazel/bazel_73/default.nix

```diff
@@ -0,0 +1,697 @@
+{
+  stdenv,
+  # nix tooling and utilities
+  lib,
+  fetchurl,
+  makeWrapper,
+  writeTextFile,
+  substituteAll,
+  writeShellApplication,
+  makeBinaryWrapper,
+  autoPatchelfHook,
+  buildFHSEnv,
+  # this package (through the fixpoint glass)
+  # TODO probably still need for tests at some point
+  bazel_self,
+  # native build inputs
+  runtimeShell,
+  zip,
+  unzip,
+  bash,
+  coreutils,
+  which,
+  gawk,
+  gnused,
+  gnutar,
+  gnugrep,
+  gzip,
+  findutils,
+  diffutils,
+  gnupatch,
+  file,
+  installShellFiles,
+  lndir,
+  python3,
+  # Apple dependencies
+  cctools,
+  libcxx,
+  sigtool,
+  CoreFoundation,
+  CoreServices,
+  Foundation,
+  IOKit,
+  # Allow to independently override the jdks used to build and run respectively
+  buildJdk,
+  runJdk,
+  # Always assume all markers valid (this is needed because we remove markers; they are non-deterministic).
+  # Also, don't clean up environment variables (so that NIX_ environment variables are passed to compilers).
+  enableNixHacks ? false,
+  version ? "7.3.0",
```

Comment:
We might want to target [7.3.1](https://github.com/bazelbuild/bazel/releases/tag/7.3.1) immediately since it contains bug fixes and should not require too many changes.

Reply:
I think you are correct. I will get that started. Hopefully it's nothing more than updating some hashes. Bazel has been known to break compatibility under the hood in a "patch" release, but I think this one should be low risk.

Reply:
This is now targeting 7.3.1.